### ISSUE TITLE
FactorizationMachine implementation and paper are different

### ISSUE THREAD

**Issue by JavanTang** [rixwew/pytorch-fm#22]
```
class FactorizationMachine(torch.nn.Module):

    def __init__(self, reduce_sum=True):
        super().__init__()
        self.reduce_sum = reduce_sum

    def forward(self, x):
        """
        :param x: Float tensor of size ``(batch_size, num_fields, embed_dim)``
        """
        square_of_sum = torch.sum(x, dim=1) ** 2
        sum_of_square = torch.sum(x ** 2, dim=1)
        ix = square_of_sum - sum_of_square
        if self.reduce_sum:
            ix = torch.sum(ix, dim=1, keepdim=True)
        return 0.5 * ix
```

![image](https://user-images.githubusercontent.com/25953568/84103525-df0f2b00-aa45-11ea-80a3-289691b9b74e.png)

What happened to v parameters?

**Comment by yfreedomliTHU** [comment]
> ```
> class FactorizationMachine(torch.nn.Module):
> 
>     def __init__(self, reduce_sum=True):
>         super().__init__()
>         self.reduce_sum = reduce_sum
> 
>     def forward(self, x):
>         """
>         :param x: Float tensor of size ``(batch_size, num_fields, embed_dim)``
>         """
>         square_of_sum = torch.sum(x, dim=1) ** 2
>         sum_of_square = torch.sum(x ** 2, dim=1)
>         ix = square_of_sum - sum_of_square
>         if self.reduce_sum:
>             ix = torch.sum(ix, dim=1, keepdim=True)
>         return 0.5 * ix
> ```
> 
> ![image](https://user-images.githubusercontent.com/25953568/84103525-df0f2b00-aa45-11ea-80a3-289691b9b74e.png)
> 
> What happened to v parameters?

class FactorizationMachineModel(torch.nn.Module):
    """
    A pytorch implementation of Factorization Machine.
    Reference:
        S Rendle, Factorization Machines, 2010.
    """

    def __init__(self, field_dims, embed_dim):
        super().__init__()
        self.embedding = FeaturesEmbedding(field_dims, embed_dim)
        self.linear = FeaturesLinear(field_dims)
        self.fm = FactorizationMachine(reduce_sum=True)

    def forward(self, x):
        """
        :param x: Long tensor of size ``(batch_size, num_fields)``
        """
        x = self.linear(x) + self.fm(self.embedding(x))
        return torch.sigmoid(x.squeeze(1))

The implementation of FM is in fm.py, which is actually class FactorizationMachineModel. And Parameters V is implemented by FeaturesEmbedding.

**Comment by JavanTang** [comment]
I see, thank you
